src/libs/ExportOnyxState/common.ts

```diff
@@ -28,7 +57,20 @@ const maskFragileData = (data: Record<string, unknown> | unknown[] | null, paren
     }
 
     if (Array.isArray(data)) {
-        return data.map((item): unknown => (typeof item === 'object' ? maskFragileData(item as Record<string, unknown>, parentKey) : item));
+        return data.map((item): unknown => {
+            if (typeof item === 'string' && Str.isValidEmail(item)) {
+                let maskedEmail = '';
+                if (!emailMap.has(item)) {
+                    maskedEmail = randomizeEmail(item);
+                    emailMap.set(item, maskedEmail);
+                } else {
+                    // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+                    maskedEmail = emailMap.get(item) as string;
+                }
+                return maskedEmail;
```

Comment:
I think we can wrap these lines in a new method `getMarkedEmail` and reuse it, like

```ts
const getMarkedEmail = (email: string): string => {
    let maskedEmail = '';
    if (!emailMap.has(email)) {
        maskedEmail = randomizeEmail(email);
        emailMap.set(email, maskedEmail);
    } else {
        // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
        maskedEmail = emailMap.get(email) as string;
    }
    return maskedEmail;
};
```